src/Client/Connection.cpp

```diff
@@ -860,6 +874,7 @@ void Connection::sendQuery(
             data += query;
             data += query_id;
             data += client_info->initial_user;
+            data += external_roles_str;
```

Comment:
Hm, maybe we can then remove this line here?

We can calculate hash and ignore external roles. Or this is a bad solution? Why?

Reply:
No, we should keep this because otherwise, you can forge a malicious request with privilege escalations.
I think the solution is just to add another `if (server_revision >= DBMS_MIN_PROTOCOL_VERSION_WITH_INTERSERVER_EXTERNALLY_GRANTED_ROLES)` for this line.